x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -443,7 +448,22 @@ public LoggingAuditTrail(Settings settings, ClusterService clusterService, Threa
     }
 
     @Override
-    public void authenticationSuccess(String requestId, Authentication authentication, RestRequest request) {
+    public void authenticationSuccess(RestRequest request) {
+        final String requestId = AuditUtil.extractRequestId(securityContext.getThreadContext());
+        if (requestId == null) {
+            throw new ElasticsearchSecurityException("Authenticated context must include request id");
+        }
+        final Authentication authentication;
+        try {
+            authentication = securityContext.getAuthentication();
+        } catch (Exception e) {
+            logger.error(() -> format("caught exception while trying to read authentication from request [%s]", request), e);
+            tamperedRequest(requestId, request);
```

Comment:
For my own education ... what exactly is a tampered request ?  I thought it happened for some reason you can not deserialize the context needed ... If so, I don't think this is the same since there are other reasons you may get an exception here. Perhaps introduce a new Exception type and handle that specifically ?